bika/lims/browser/fields/uidreferencefield.py

```
@@ -0,0 +1,193 @@
+        :type kwargs: dict
+        :return: object or list of objects for multiValued fields.
+        :rtype: BaseContent | list[BaseContent]
+        """
+        value = StringField.get(self, context, **kwargs)
+        if self.multiValued:
+            ret = value
+        else:
+            ret = self.get_object(context, value)
+            if aslist:
+                ret = [ret]
+        return ret
+
+    @security.public
+    def set(self, context, value, **kwargs):
+        """Accepts a UID, brain, or an object (or a list of any of these),
+        and stores a UID or list of UIDS.
+
+        :param context: context is the object who's schema contains this field.
+        :type context: BaseContent
+        :param value: A UID, brain or object (or a sequence of these).
+        :type value: Any
+        :param kwargs: kwargs are passed directly to the underlying get.
+        :type kwargs: dict
+        :return: object or list of objects for multiValued fields.
+        :rtype: BaseContent | list[BaseContent]
+        """
+        if self.multiValued:
+            if type(value) not in (list, tuple):
+                value = [value, ]
+            ret = []
+            # filter out duplicates
+            for val in value:
+                if val not in ret:
+                    ret.append(self.to_uid(context, val))
+        else:
+            # Sometimes we get given a list here with an empty string.
+            # This is generated by html forms with empty values.
+            # This is a single-valued field though, so:
+            if isinstance(value, (list, tuple)) and value:
+                if len(value) > 1:
+                    logger.warning(
+                        "Found values '\'{}\'' for singleValued field <{}>.{} "
+                        "- using only the first value in the list.".format(
+                            '\',\''.join(value), context.UID(), self.getName()))
+                value = value[0] if value else ""
```

I would rather raise an Error than falling back to the first element of the list.
Write the whole transaction or nothing...